src/app/(landing)/leaderboard/page.tsx

```diff
@@ -0,0 +1,118 @@
+"use client"
+
+import { useCallback, useMemo, useRef } from "react"
+
+import LeaderboardHighlight from "~/components/leaderboard-screen/LeaderboardHighlight"
+import { type Doc } from "~/lib/db/types"
+import { api } from "~/lib/trpc/react"
+import { cn } from "~/lib/utils"
+
+export default function LeaderboardPage() {
+  const observer = useRef<IntersectionObserver>()
+
+  const infiniteLeaderboard = api.leaderboard.getLeaderboard.useInfiniteQuery(
+    { limit: 10 },
+    {
+      getNextPageParam: (lastPage) => (lastPage.cursor !== null ? lastPage.cursor + 10 : undefined),
+    },
+  )
+
+  const lastElementRef = useCallback(
+    (node: HTMLTableRowElement) => {
+      if (infiniteLeaderboard.isLoading) return
+
+      if (observer.current) observer.current.disconnect()
+
+      observer.current = new IntersectionObserver((entries) => {
+        if (
+          entries[0]?.isIntersecting &&
+          infiniteLeaderboard.hasNextPage &&
+          !infiniteLeaderboard.isFetching
+        ) {
+          void infiniteLeaderboard.fetchNextPage()
+        }
+      })
+
+      if (node) observer.current.observe(node)
+    },
+    [infiniteLeaderboard],
+  )
+
+  const players = useMemo(() => {
+    return infiniteLeaderboard.data?.pages.reduce((acc, page) => {
+      return [...acc, ...page.users]
+    }, [] as Doc<"users">[])
+  }, [infiniteLeaderboard])
+
+  if (infiniteLeaderboard.isLoading || !players) return <h1>loading</h1>
+
+  if (infiniteLeaderboard.error) return <h1>errroed</h1>
+
+  return (
+    <div>
+      <LeaderboardHighlight players={[players[0], players[1], players[2]]} />
```

Comment:
```suggestion
      <LeaderboardHighlight players={players.slice(0, 3)} />
```

Reply:
@soorria is there a way to do that but satisfy the type of `type Players = [User, User, User]`, since I want to guarantee its a list of 3 users. If I do the slice it fails the type check. Feel like maybe I should just assume its a list of players and not rely on the fact its just of len 3